filebeat/input/file/state.go

```diff
@@ -49,5 +72,8 @@ func (s *State) IsEqual(c *State) bool {
 
 // IsEmpty returns true if the state is empty
 func (s *State) IsEmpty() bool {
-	return *s == State{}
+	return s.FileStateOS == file.StateOS{} &&
```

Comment:
Can you share some details here on why not only add s.Meta but also the other chnages?

Reply:
As I added a map to the `State` struct, a simple comparison is no longer possible (Go doesn't implement map compare), I had to update the code to do an individual comparison on the struct fields